### PR TITLE
fix hpu storage serialization

### PR DESCRIPTION
**Pull request by ppiskorski**
Change-Id: Ia534400a0e8972590374eceba5b62a2525b796e5

Fixes #ISSUE_NUMBER

cc @mruberry @mikaylagawarecki

### CODE REVIEW

File: torch/_utils.py [hunk 1]
```diff
@@ -48,6 +48,38 @@ def _type(self, dtype=None, non_blocking=False, **kwargs):
...
+        device (int): The destination HPU id. Defaults to the current device.
+        non_blocking (bool): If ``True`` and the source is in pinned memory,
+            the copy will be asynchronous with respect to the host. Otherwise,
+            the argument has no effect.
+        **kwargs: For compatibility, may contain the key ``async`` in place of
+            the ``non_blocking`` argument.
+    """
+    non_blocking = _get_async_or_non_blocking("hpu", non_blocking, kwargs)
+    hpu = getattr(torch, "hpu", None)
+    assert hpu is not None, "HPU device module is not loaded"
```
Comment (mikaylagawarecki) [inline torch/_utils.py:67]: is it guaranteed that hpu will be available if hasattr(torch, "hpu")? do we have an equivalent of `torch.{device}.is_available()` for hpu?

File: torch/serialization.py [hunk 2]
```diff
@@ -198,6 +202,38 @@ def _cuda_deserialize(obj, location):
             return obj.cuda(device)
 
 
+def validate_hpu_device(location):
+    hpu = getattr(torch, "hpu", None)
+    assert hpu is not None, "HPU device module is not loaded"
+    device = hpu._utils._get_device_index(location)
```
Comment (mikaylagawarecki) [inline torch/serialization.py:208]: For my understanding is there a reason why we don't pass optional=True here? (just comparing with `validate_cuda_device` and curious to know if this is intentional)

**Review by mikaylagawarecki — COMMENTED** [review]
This looks good to me, just two questions

File: torch/_utils.py (hunk 1 above)
Comment (ppiskorski) [reply torch/_utils.py:67]: Actually, availability of  torch.hpu only means that the module has been loaded and subsequent check for actual device can be done with torch.hpu.is_available, just like you suspect. Only that at this level (_utils.py) there are no such checks.

Comment (mikaylagawarecki) [reply torch/_utils.py:67]: hm why is it not available in `_utils.py`? I see that `torch.xpu.is_available()` is used in this file

**Review by mikaylagawarecki — APPROVED** [review]
Thanks, this lgtm! Just one question

File: torch/_utils.py (hunk 1 above)
Comment (mikaylagawarecki) [reply torch/_utils.py:67]: Got it, thanks for clarifying

**Pull request closed without merging**